.github/workflows/maven.yml

```diff
@@ -34,8 +36,8 @@ jobs:
       run: mvn -Dtest=SparkRuntimeTests test
     - name: JavaAPITest
       run: mvn -Dtest=JavaAPITest test
-    - name: Spotless check
-      run: mvn spotless:check
+    - name: Install Spark and run server tests
+      run: mkdir -p /opt; wget -q -O /opt/spark.tgz https://downloads.apache.org/spark/spark-2.4.5/spark-2.4.5-bin-hadoop2.7.tgz ; tar -xzf /opt/spark.tgz -C /opt/ ; rm /opt/spark.tgz ; export SPARK_HOME=/opt/spark-2.4.5-bin-hadoop2.7 ; export PATH=$PATH:/opt/spark-2.4.5-bin-hadoop2.7/bin; bash server-test-script.sh
```

Comment:
Do you think we could split this into several intermediate step for ease of read?

Reply:
For readability I have initially aimed to use multi-line commands which are supposedly supported by workflow .yml. However, when tested, it would only run the first command and skip the rest. So I have opted for chaining them via semi-colons.

The reason I have put the entire Spark installation and testing in a single step is to make it clear that this is the only step that uses the Spark installation. 

I will nonetheless, go over the alternatives once more to make this more readable. I would be glad to hear any preferences you may have(no matter weak or strong) on how to structure this.